src/tms_service.ts

```diff
@@ -0,0 +1,337 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import _ from 'lodash';
+import { ORIGIN } from './origin';
+import { toAbsoluteUrl } from './utils';
+import { EMSClient, ITMSService, EmsLayerAttribution } from './ems_client';
+import { Sources, Style, VectorSource } from 'mapbox-gl';
+
+interface EmsVectorSource extends VectorSource {
```

Comment:
for there "bag-of-property" key-value types, we've been using type-aliases rather than interfaces https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-aliases in Maps.

The reason for this is that it's closer to a struct-like data object, rather than the definition of some behavior (which seems more in line with interfaces).